cmake/singularity/Eigen3.cmake

```diff
@@ -0,0 +1,5 @@
+macro(singularityeos_enable_eigen3 target)
+  find_package(Eigen3 REQUIRED)
+  target_link_libraries(${target} PUBLIC Eigen3::Eigen)
+  set(SINGULARITY_EOS_ENABLE_MPARK-VARIANT ON)
+endmacro()
```

Comment:
I'm fine either way, but I'm curious: why a per-file macro here rather than just doing this stuff inline? Just easier from an organization/readability perspective?

Reply:
Just makes it easier to manage. Some packages require a lot of setup/configuration so it's easier to isolate all that into a file. Ones with shorter macros could be inline, but it makes it clear where the code for doing the dependency stuff is (e.g., if something fails with `find_package`, the stack will show `cmake/singularity/<package>.cmake`).

Also, in future we can put more "package-specific" stuff here, like options or maintaining version differences.